static/src/javascripts/projects/commercial/modules/dfp/get-ad-iframe.ts

```diff
@@ -0,0 +1,47 @@
+interface HTMLIFrameElementIE extends HTMLIFrameElement {
+	readyState: string | undefined;
+}
+
+const getAdvertIframe = (adSlot: HTMLElement) =>
+	new Promise((resolve, reject) => {
+		// DFP will sometimes return empty iframes, denoted with a '__hidden__' parameter embedded in its ID.
+		// We need to be sure only to select the ad content frame.
+		const contentFrame = adSlot.querySelector<HTMLIFrameElement>(
+			'iframe:not([id*="__hidden__"])',
+		);
+
+		if (!contentFrame) {
+			reject();
+		} else if (
+			// According to Flow, readyState exists on the Document, not the HTMLIFrameElement
+			// Is this different for old IE?
```

Comment:
Is this comment left over from the old flow javascript, does it need changing it now we've converted to TS, or does it still apply?

Reply:
It's still valid: `readyState` on an iframe seems to be an IE thing. I can update it.